CONTRIBUTING.md

```diff
@@ -3,9 +3,12 @@
 ### Getting and building
 
 ### Assumed
- * A working GCC (on mac os x something like `xcode-select --install` will get you started)
- * [Go environment](http://golang.org/doc/code.html)
- * Git and Mercurial (for retrieving dependencies)
+ * A working C++ compiler (on mac os x something like `xcode-select
+   --install` will get you started). The compiler must support C++11
+   (GCC 4.9+ and clang 3.6+ are known to work).
+ * [Go environment](http://golang.org/doc/code.html). Currently a
+   64-bit version of go is required.
```

Comment:
Perhaps we should check for some of these preconditions in the makefile. For example, it wouldn't be difficult to check the Go version and whether it is 64-bit using `go version`. Similarly, we could determine what CC and CXX compiler Go is using (`go env CC CXX`) and verify they are recent enough.